driver/src/it/scala/org/mongodb/scala/DocumentationExampleSpec.scala

```diff
@@ -0,0 +1,516 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mongodb.scala
+
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.language.implicitConversions
+
+import org.mongodb.scala.bson.{BsonArray, BsonDocument, BsonString}
+
+//scalastyle:off magic.number
+class DocumentationExampleSpec extends RequiresMongoDBISpec {
+
+  // Implicit functions that execute the Observable and return the results
+  val waitDuration = Duration(5, "seconds")
+  implicit class ObservableExecutor[T](observable: Observable[T]) {
+    def execute(): Seq[T] = Await.result(observable, waitDuration)
+  }
+
+  implicit class SingleObservableExecutor[T](observable: SingleObservable[T]) {
+    def execute(): T = Await.result(observable, waitDuration)
+  }
+
+  "The Scala driver" should "be able to insert" in withCollection { collection =>
+
+    // Start Example 1
+    collection.insertOne(
+      Document("""{ item: "canvas", qty: 100, tags: ["cotton"], size: { h: 28, w: 35.5, uom: "cm" } }""")
+    ).execute()
+    // End Example 1
+
+    // Start Example 2
+    val observable = collection.find(Document("{item: 'canvas'}"))
+    // End Example 2
+
+    observable.execute().size should equal(1)
+
+    // Start Example 3
+    collection.insertMany(Seq(
+      Document("""{ item: "journal", qty: 25, tags: ["blank", "red"], size: { h: 14, w: 21, uom: "cm" } }"""),
+      Document("""{ item: "mat", qty: 85, tags: ["gray"], size: { h: 27.9, w: 35.5, uom: "cm" } }"""),
+      Document("""{ item: "mousepad", qty: 25, tags: ["gel", "blue"], size: { h: 19, w: 22.85, uom: "cm" } }""")
+    )).execute()
+    // End Example 3
+
+    collection.count().execute() should equal(4)
+  }
+
+  it should "be able to query top level" in withCollection { collection =>
+
+    // Start Example 6
+    collection.insertMany(Seq(
+      Document("""{ item: "journal", qty: 25, size: { h: 14, w: 21, uom: "cm" }, status: "A" }"""),
+      Document("""{ item: "notebook", qty: 50, size: { h: 8.5, w: 11, uom: "in" }, status: "A" }"""),
+      Document("""{ item: "paper", qty: 100, size: { h: 8.5, w: 11, uom: "in" }, status: "D" }"""),
+      Document("""{ item: "planner", qty: 75, size: { h: 22.85, w: 30, uom: "cm" }, status: "D" }"""),
+      Document("""{ item: "postcard", qty: 45, size: { h: 10, w: 15.25, uom: "cm" }, status: "A" }""")
+    )).execute()
+    // End Example 6
+
+    collection.count().execute() should equal(5)
+
+    // Start Example 7
+    var findObservable = collection.find(Document())
+    // End Example 7
+
+    findObservable.execute().size should equal(5)
+
+    // Start Example 8
+    findObservable = collection.find()
+    // End Example 8
+
+    findObservable.execute().size should equal(5)
+
+    // Start Example 9
+    import org.mongodb.scala.model.Filters
+    findObservable = collection.find(Filters.eq("status", "D"))
```

Comment:
It would be a shame that all Updates, Projections, and Filters have to be spelled out because of a conflict with the testing library for just one (or perhaps a few) of those methods, especially since the assertions are not included in the resulting documentation (though I understand why we want them).

Any alternatives you can think of?  If the problem is with the Matchers trait, is there a more verbose alternative that doesn't conflict?